scalabel/label/to_rle.py

```diff
@@ -0,0 +1,141 @@
+    parser.add_argument(
+        "-i",
+        "--input",
+        help=(
+            "root directory of scalabel Json files or path to a label "
+            "json file"
+        ),
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        help="path to save rle formatted label file",
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        default="det",
+        choices=["ins_seg", "sem_seg", "pan_seg", "seg_track"],
+        help="conversion mode",
+    )
+    parser.add_argument(
+        "--nproc",
+        type=int,
+        default=NPROC,
+        help="number of processes for conversion",
+    )
+    parser.add_argument(
+        "--config",
+        type=str,
+        default=None,
+        help="Configuration file",
+    )
+    return parser.parse_args()
+
+
+def frame_to_rle(shape: ImageSize, frame: Frame) -> Frame:
+    """Converting a frame of poly2ds to rle."""
+    labels = frame.labels
+    if labels is None or len(labels) == 0:
+        return frame
+    # higher score, rendering later
+    has_score = all((label.score is not None for label in labels))
+    if has_score:
+        labels = sorted(
+            labels, key=lambda label: float(label.score)  # type: ignore
+        )
+    poly2ds = [label.poly2d for label in labels if label.poly2d is not None]
+    rles = frame_to_rles(shape, poly2ds)
+    for label, rle in zip(labels, rles):
+        label.rle = rle
+        label.box2d = rle_to_box2d(rle)
+    return frame
+
+
+def frames_to_rle(
+    shapes: List[ImageSize],
+    frames: List[Frame],
+    nproc: int = NPROC,
+) -> List[Frame]:
+    """Execute the rle conversion in parallel."""
+    with Pool(nproc) as pool:
+        frames = pool.starmap(
+            partial(frame_to_rle),
+            tqdm(
+                zip(shapes, frames),
+                total=len(frames),
+            ),
+        )
+
+    sorted(frames, key=lambda frame: frame.name)
+    return frames
+
+
+def seg_to_rles(
+    frames: List[Frame], config: Config, nproc: int = NPROC
+) -> List[Frame]:
+    """Converting segmentation poly2d to rles."""
+    img_shape = config.imageSize
+    assert img_shape is not None, "Seg conversion requires imageSize in config"
+    img_shapes = [img_shape] * len(frames)
+    logger.info("Start conversion for Seg to RLEs")
+    return frames_to_rle(img_shapes, frames, nproc)
+
+
+def main() -> None:
+    """Main function."""
+    args = parse_args()
+    assert args.mode in ["ins_seg", "sem_seg", "pan_seg", "seg_track"]
+
+    dataset = load(args.input, args.nproc)
+    frames, config = dataset.frames, dataset.config
+
+    if args.config is not None:
+        config = load_label_config(args.config)
+    if config is None:
+        raise ValueError(
+            "Dataset config is not specified. Please use --config"
+            " to specify a config for this dataset."
+        )
+
+    frames = seg_to_rles(frames, config, args.nproc)
+    if args.mode == "seg_track":
```

Comment:
there is no necessity to store a json per video, so i think mode 'seg_track' is not accurate here. Rather something like --json-per-video if applicable. So my point is, whether to save the modified frames in a single json or json per video does not depend on the dataset type but on the user's preference, hence the mode option andthe error message in L130 seems inaccurate

Reply:
That's a good point, I'll make it a command-line option then